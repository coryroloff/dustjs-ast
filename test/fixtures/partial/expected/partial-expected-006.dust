{> "thing" myparam=443.2 /}