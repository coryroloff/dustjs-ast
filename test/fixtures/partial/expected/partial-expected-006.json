{
	"type": "Template",
	"body": [
		{
			"type": "Partial",
			"value": {
				"type": "Inline",
				"body": [
					{
						"type": "StringLiteral",
						"value": "thing",
						"loc": {
							"start": {
								"line": 1,
								"column": 5
							},
							"end": {
								"line": 1,
								"column": 10
							}
						}
					}
				],
				"loc": {
					"start": {
						"line": 1,
						"column": 4
					},
					"end": {
						"line": 1,
						"column": 11
					}
				}
			},
			"context": null,
			"params": [
				{
					"type": "Param",
					"key": {
						"type": "Key",
						"name": "myparam",
						"loc": {
							"start": {
								"line": 1,
								"column": 12
							},
							"end": {
								"line": 1,
								"column": 19
							}
						}
					},
					"value": {
						"type": "Inline",
						"body": [
							{
								"type": "StringLiteral",
								"value": "start",
								"loc": {
									"start": {
										"line": 1,
										"column": 21
									},
									"end": {
										"line": 1,
										"column": 26
									}
								}
							},
							{
								"type": "Reference",
								"id": {
									"type": "Identifier",
									"path": [
										{
											"type": "Key",
											"name": "ref",
											"loc": {
												"start": {
													"line": 1,
													"column": 27
												},
												"end": {
													"line": 1,
													"column": 30
												}
											}
										}
									],
									"localContext": false,
									"loc": {
										"start": {
											"line": 1,
											"column": 27
										},
										"end": {
											"line": 1,
											"column": 30
										}
									}
								},
								"filters": [],
								"loc": {
									"start": {
										"line": 1,
										"column": 26
									},
									"end": {
										"line": 1,
										"column": 31
									}
								}
							},
							{
								"type": "StringLiteral",
								"value": "end",
								"loc": {
									"start": {
										"line": 1,
										"column": 31
									},
									"end": {
										"line": 1,
										"column": 34
									}
								}
							}
						],
						"loc": {
							"start": {
								"line": 1,
								"column": 20
							},
							"end": {
								"line": 1,
								"column": 35
							}
						}
					},
					"loc": {
						"start": {
							"line": 1,
							"column": 12
						},
						"end": {
							"line": 1,
							"column": 35
						}
					}
				}
			],
			"loc": {
				"start": {
					"line": 1,
					"column": 1
				},
				"end": {
					"line": 1,
					"column": 38
				}
			}
		}
	],
	"tokens": [
		{
			"type": "Punctuator",
			"value": "{",
			"loc": {
				"start": {
					"line": 1,
					"column": 1
				},
				"end": {
					"line": 1,
					"column": 2
				}
			}
		},
		{
			"type": "Punctuator",
			"value": ">",
			"loc": {
				"start": {
					"line": 1,
					"column": 2
				},
				"end": {
					"line": 1,
					"column": 3
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "\"",
			"loc": {
				"start": {
					"line": 1,
					"column": 4
				},
				"end": {
					"line": 1,
					"column": 5
				}
			}
		},
		{
			"type": "StringLiteral",
			"value": "thing",
			"loc": {
				"start": {
					"line": 1,
					"column": 5
				},
				"end": {
					"line": 1,
					"column": 10
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "\"",
			"loc": {
				"start": {
					"line": 1,
					"column": 10
				},
				"end": {
					"line": 1,
					"column": 11
				}
			}
		},
		{
			"type": "Key",
			"value": "myparam",
			"loc": {
				"start": {
					"line": 1,
					"column": 12
				},
				"end": {
					"line": 1,
					"column": 19
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "=",
			"loc": {
				"start": {
					"line": 1,
					"column": 19
				},
				"end": {
					"line": 1,
					"column": 20
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "\"",
			"loc": {
				"start": {
					"line": 1,
					"column": 20
				},
				"end": {
					"line": 1,
					"column": 21
				}
			}
		},
		{
			"type": "StringLiteral",
			"value": "start",
			"loc": {
				"start": {
					"line": 1,
					"column": 21
				},
				"end": {
					"line": 1,
					"column": 26
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "{",
			"loc": {
				"start": {
					"line": 1,
					"column": 26
				},
				"end": {
					"line": 1,
					"column": 27
				}
			}
		},
		{
			"type": "Key",
			"value": "ref",
			"loc": {
				"start": {
					"line": 1,
					"column": 27
				},
				"end": {
					"line": 1,
					"column": 30
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "}",
			"loc": {
				"start": {
					"line": 1,
					"column": 30
				},
				"end": {
					"line": 1,
					"column": 31
				}
			}
		},
		{
			"type": "StringLiteral",
			"value": "end",
			"loc": {
				"start": {
					"line": 1,
					"column": 31
				},
				"end": {
					"line": 1,
					"column": 34
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "\"",
			"loc": {
				"start": {
					"line": 1,
					"column": 34
				},
				"end": {
					"line": 1,
					"column": 35
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "/",
			"loc": {
				"start": {
					"line": 1,
					"column": 36
				},
				"end": {
					"line": 1,
					"column": 37
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "}",
			"loc": {
				"start": {
					"line": 1,
					"column": 37
				},
				"end": {
					"line": 1,
					"column": 38
				}
			}
		}
	],
	"loc": {
		"start": {
			"line": 1,
			"column": 1
		},
		"end": {
			"line": 1,
			"column": 38
		}
	}
}